File: src/main/java/com/tronic/bot/storage/StorageElement.java
package com.tronic.bot.storage;

import com.toddway.shelf.Shelf;
import kotlin.jvm.JvmClassMappingKt;

import java.io.Serializable;
import java.util.List;

public abstract class StorageElement {

    Shelf shelf;

    StorageElement(Shelf shelf) {
        this.shelf = shelf;
    }

    void set(String item,Object value) {

        this.shelf.item(item).put(value);
    }

     Object get(String item,Class value) {
        Shelf.Item itemm = this.shelf.item(item);
        return itemm.get(JvmClassMappingKt.getKotlinClass(value));
    }

    List<Object> getList(String item,Class classes) {
        return this.shelf.item(item).getList(JvmClassMappingKt.getKotlinClass(classes));
    }

}

File: src/main/java/com/tronic/bot/storage/StorageElement.kt
package com.tronic.bot.storage;

import com.toddway.shelf.Shelf;
import com.toddway.shelf.get
import kotlin.reflect.KClass


abstract class StorageElement(private val shelf: Shelf) {

    fun set(item: String,value: Any) {
        this.shelf.item(item).put(value);
    }

     fun <T: Any> get(item: String,value: KClass<T>): T? {
        val itemm:Shelf.Item = this.shelf.item(item);
        return itemm.get(value)
    }

    fun <T: Any> getList(item: String,classes: KClass<T>): List<T>? {
        return this.shelf.item(item).getList(classes)
    }

}
